public/app/features/alerting/unified/Silences.test.tsx

```diff
@@ -0,0 +1,260 @@
+    durationField: byLabelText('Duration'),
+    durationInput: byRole('textbox', { name: /duration/i }),
+    matchersField: byTestId('matcher'),
+    matcherName: byPlaceholderText('label'),
+    matcherValue: byPlaceholderText('value'),
+    comment: byPlaceholderText('Details about the silence'),
+    createdBy: byPlaceholderText('Username'),
+    matcherOperatorSelect: byLabelText('operator'),
+    matcherOperator: (operator: MatcherOperator) => byText(operator, { exact: true }),
+    addMatcherButton: byRole('button', { name: 'Add matcher' }),
+    submit: byText('Submit'),
+  },
+};
+
+const resetMocks = () => {
+  mocks.api.fetchSilences.mockImplementation(() => {
+    return Promise.resolve([
+      mockSilence({ id: '12345' }),
+      mockSilence({ id: '67890', matchers: parseMatchers('foo!=bar'), comment: 'Catch all' }),
+    ]);
+  });
+
+  mocks.api.fetchAlerts.mockImplementation(() => {
+    return Promise.resolve([
+      mockAlertmanagerAlert({
+        labels: { foo: 'bar' },
+        status: { state: AlertState.Suppressed, silencedBy: ['12345'], inhibitedBy: [] },
+      }),
+      mockAlertmanagerAlert({
+        labels: { foo: 'buzz' },
+        status: { state: AlertState.Suppressed, silencedBy: ['67890'], inhibitedBy: [] },
+      }),
+    ]);
+  });
+
+  mocks.api.createOrUpdateSilence.mockResolvedValue(mockSilence());
+};
+
+describe('Silences', () => {
+  beforeAll(resetMocks);
+  afterEach(resetMocks);
+
+  beforeEach(() => {
+    setDataSourceSrv(new MockDataSourceSrv(dataSources));
+  });
+
+  it('loads and shows silences', async () => {
+    renderSilences();
+    await waitFor(() => expect(mocks.api.fetchSilences).toHaveBeenCalled());
+    await waitFor(() => expect(mocks.api.fetchAlerts).toHaveBeenCalled());
+
+    expect(ui.silencesTable.query()).not.toBeNull();
+    expect(ui.silenceRow.queryAll()).toHaveLength(2);
+  });
+
+  it('shows the correct number of silenced alerts', async () => {
+    mocks.api.fetchAlerts.mockImplementation(() => {
+      return Promise.resolve([
+        mockAlertmanagerAlert({
+          labels: { foo: 'bar', buzz: 'bazz' },
+          status: { state: AlertState.Suppressed, silencedBy: ['12345'], inhibitedBy: [] },
+        }),
+        mockAlertmanagerAlert({
+          labels: { foo: 'bar', buzz: 'bazz' },
+          status: { state: AlertState.Suppressed, silencedBy: ['12345'], inhibitedBy: [] },
+        }),
+      ]);
+    });
+
+    renderSilences();
+    await waitFor(() => expect(mocks.api.fetchSilences).toHaveBeenCalled());
+    await waitFor(() => expect(mocks.api.fetchAlerts).toHaveBeenCalled());
+
+    const silencedAlertRows = ui.silencedAlertCell.getAll(ui.silencesTable.get());
+    expect(silencedAlertRows).toHaveLength(2);
+    expect(silencedAlertRows[0]).toHaveTextContent('2');
+    expect(silencedAlertRows[1]).toHaveTextContent('0');
+  });
+
+  it('filters silences by matchers', async () => {
+    renderSilences();
```

Comment:
this should be `await`ed

Reply:
vs code tells me that `await` will have no effect on that function, is `render` async?

Reply:
oh, my bad! Somehow I was sure `render` returns a promise :